WMF Framework/ReadingListsSyncOperation.swift

```diff
@@ -878,19 +882,31 @@ internal class ReadingListsSyncOperation: ReadingListsOperation, @unchecked Send
                 entry.articleKey = article.key
                 entry.variant = article.variant
                 entry.displayTitle = article.displayTitle
+
                 if article.savedDate == nil {
                     article.savedDate = entry.createdDate as Date?
                 }
+
+                if needsMigrationToWMFData, let url = article.url {
+                    newlySavedArticleIDs.insert(url)
+                }
+
                 updatedLists.insert(readingList)
             }
         }
-        for readingList in updatedLists {
+        for readingList in updatedLists {entryCount += 1
             try autoreleasepool {
                 try readingList.updateArticlesAndEntries()
             }
         }
+
+        // Registered newly synced saved articles to WMFData
+        if !newlySavedArticleIDs.isEmpty {
```

Comment:
I saw in an old [writeup](https://phabricator.wikimedia.org/T400371#11070198) of mine that articles could be deleted via a sync as well, so that is worth testing. It's possible that the moment it happens is [here](https://github.com/wikimedia/wikipedia-ios/blob/main/Wikipedia/Code/ReadingListsController.swift#L734) (that is the only area in the codebase where I can find `savedDate = nil`).